packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -73,7 +45,45 @@ export async function packageLayer(context, resource) {
   });
 }
 
-export async function hashLayerDir(layerPath: string): Promise<any> {
+// Check hash results for content changes, bump version if so
+async function ensureLayerVersion(context: any, layerPath: string, layerName: string) {
+  const layerData = getLayerMetadataFactory(context)(layerName);
+  let latestVersion: number = layerData.getLatestVersion();
+  const { hash: currentHash } = await hashLayerDir(layerPath);
+  const previousHash = layerData.getHash(latestVersion);
+  const layerParameters = context.amplify.readJsonFile(path.join(layerPath, 'layer-parameters.json'));
+
+  if (previousHash && previousHash !== currentHash) {
+    const prevPermissions = layerData.getVersion(latestVersion).permissions;
+    ++latestVersion; // Content changes detected, bumping version
+    layerParameters.layerVersionMap[`${latestVersion}`] = {
+      permissions: await getNewVersionPermissions(layerName, prevPermissions),
+      hash: currentHash,
+    };
+    createLayerParametersFile(context, layerParameters, layerPath);
+    layerParameters.layerName = layerName;
+    layerParameters.build = true;
+    updateLayerCfnFile(context, layerParameters, layerPath);
+  } else if (!previousHash) {
+    layerParameters.layerVersionMap[`${latestVersion}`].hash = currentHash;
+    createLayerParametersFile(context, layerParameters, layerPath);
+  }
+}
+
+async function getNewVersionPermissions(
+  layerName: string,
+  prevPermissions: Partial<LayerPermission>[],
+): Promise<Partial<LayerPermission>[]> {
+  const defaultPermissions: PrivateLayer[] = [{ type: Permission.private }];
+  let usePrevPermissions = true;
+  if (!_.isEqual(prevPermissions, defaultPermissions)) {
+    const { usePrevPerms } = await prompt(prevPermsQuestion(layerName));
+    usePrevPermissions = usePrevPerms === 'previous';
+  }
+  return new Promise(resolve => (usePrevPermissions ? prevPermissions : defaultPermissions));
```

Comment:
nit can just `return usePrevPermissions ? prevPermissions : defaultPermissions`. Since it's an async function, it will automagically be wrapped in a promise